File: Medium/1004_maximum_consecutive_ones_iii/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        n = len(nums)

        res = 0
        zero_cnt = 0
        l = 0
        for r in range(n):
            if nums[r] == 0:
                zero_cnt += 1
            while zero_cnt > k:
                if nums[l] == 0:
                    zero_cnt -= 1
                l += 1
            res = max(r-l+1, res)

        return res

File: Medium/1004_maximum_consecutive_ones_iii/solution.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, res = 0;
        int zero_count = 0;
        for (int r=0; r<nums.size(); r++) {
            if (nums[r] == 0) zero_count++;
            while (zero_count > k) {
                if (nums[l] == 0) zero_count--;
                l++;
            }
            res = max(r-l+1, res);
        }
        return res;
    }
};
